client/src/components/QA/QAAddAnswerModal.jsx

```diff
@@ -21,28 +13,35 @@ const QAAddAnswerModal = ({ question, productName }) => {
-        <h2>Submit Your Answer</h2>
-        <h3>{productName.name}: {question.question_body} </h3>
-        <form className="add-answer-form">
-          <label className="add-form-answer">Answer* </label>
-            <textarea rows="10" maxLength="1000" required onChange={() => setAnswer(event.target.value)}/><p></p>
-          <label className="add-form-username">Nickname*</label>
-            <input maxLength="60" placeholder="Example: jack543!" required onChange={() => setNickname(event.target.value)}/><p className="qa-form-small">For privacy reasons, do not use your full name or email address</p>
-          <label className="add-form-email">Email*</label>
-            <input maxLength="60" placeholder="Example: jack@email.com" required onChange={() => setEmail(event.target.value)}/><p className="qa-form-small">For authentication reasons, you will not be emailed</p>
-          <button className="qa-add-form-submit" onClick={() => {
-            event.preventDefault();
-            setModalIsOpenAdd(false);
-            setModalInfo({
-              answer: answer,
-              nickname: nickname,
-              email: email
-            });
-          }}>Submit</button>
-        </form>
-      </Modal>
+      <button id="qa-modal-button" className="qa-add" onClick={() => {
+        document.getElementById(index).style.display = "block";
+      }}><b>[Add Answer]</b></button>
+      <div id={index} className="qa-modal">
+        <div className="qa-modal-content">
+          <div className="qa-modal-close" onClick={() => {
+            document.getElementById(index).style.display = "none";
+          }}>&times;</div>
+          <h2>Submit Your Answer</h2>
+          <h3>{productName.name}: {question.question_body} </h3>
+          <form className="add-answer-form">
+            <label className="add-form-answer">Answer* </label>
+            <textarea rows="10" maxLength="1000" required onChange={() => setAnswer(event.target.value)} /><p></p>
+            <label className="add-form-username">Nickname*</label>
+            <input maxLength="60" placeholder="Example: jack543!" required onChange={() => setNickname(event.target.value)} /><p className="qa-form-small">For privacy reasons, do not use your full name or email address</p>
+            <label className="add-form-email">Email*</label>
+            <input maxLength="60" placeholder="Example: jack@email.com" required onChange={() => setEmail(event.target.value)} /><p className="qa-form-small">For authentication reasons, you will not be emailed</p>
+            <button className="qa-add-form-submit" onClick={() => {
+              event.preventDefault();
+              console.log('answer: ', answer, ' - nickname', nickname, ' - email: ', email);
+              setModalInfo({
+                answer: answer,
+                nickname: nickname,
+                email: email
+              });
+            }}>Submit</button>
```

Comment:
Not sure if it's in scope with this, but when the user clicks 'Submit', will it close the modal, too?

Reply:
Not at the moment, but I will work on adding it, I like this idea.